### DISCUSSION TITLE
Is there a plan for a release to support C++20?

### DISCUSSION THREAD

**Discussion by geipel** [cpputest/cpputest#1796] [General]
This library manipulates the STL to support features like memory leak detection.
It seems unlikely that this will work without changes for **C++20**.

The last release was over four years ago. Is there a plan for updated releases?

**Comment by thetic** [comment]
I believe that leak detection and GTest integration are the only features that rely on "less-than-standard" language constructs. Both of these features are optional and can be disabled.

**Reply by geipel** [reply]
Yes, we've disabled them in an existing project, which is migrating to C++20.
I probably should restate my questions differently:

- Is CppUTest being actively maintained?
- Is it known whether there are any new issues with C++20?
- Is there a plan for further releases?

**Reply by thetic** [reply]
We run tests against C++20 builds with GCC, Clang, and MSVC. It should work fine.

**Comment by basvodde** [comment]
@geipel I assume you are only talking about standard library support? The memory leak detection always clashed with standard library, so I would assume there are no changes required. Is that correct?

**Reply by geipel** [reply]
Well, I don't know.

Given that there hasn't been a release in four years, I'm wondering if this is being actively maintained.
Ultimately, I'm trying to decide if a project that's migrating to C++20 should use a different framework.
(If we find a problem, am I on my own?)

**Reply by basvodde** [reply]
It is up to you. I'm at the moment the main maintainer and I don't have time for doing releases, especially as most people just use the trunk and there are not very major changes in recent years.

We will expand the maintainers group in the future. I have no intention to completely stop maintaining it especially as it has an active usage community.

**Reply by geipel** [reply]
OK. I have not tried a version on the trunk, as we have only used released builds, historically.
If I run into any problems, I may try that.

Thanks for your responses.
